src/Microsoft.OpenApi.OData.Reader/Generator/OpenApiSchemaGenerator.cs

```diff
@@ -198,22 +224,76 @@ private static OpenApiSchema CreateCollectionSchema(ODataContext context, OpenAp
                     }
                 }
             };
-            if (context.Settings.EnablePagination)
-            {
-                properties.Add(
-                    "@odata.nextLink",
-                    new OpenApiSchema
-                    {
-                        Type = "string"
-                    });
-            }
 
-            return new OpenApiSchema
+            OpenApiSchema baseSchema = new()
             {
-                Title = $"Collection of {typeName}",
                 Type = "object",
                 Properties = properties
             };
+
+            OpenApiSchema paginationSchema = new()
+            {
+                UnresolvedReference = true,
+                Reference = new OpenApiReference
+                {
+                    Type = ReferenceType.Schema,
+                    Id = Constants.BaseCollectionPaginationResponse
+                }
+            };
+
+            OpenApiSchema countSchema = new()
+            {
+                UnresolvedReference = true,
+                Reference = new OpenApiReference
+                {
+                    Type = ReferenceType.Schema,
+                    Id = Constants.BaseCollectionCountResponse
+                }
+            };
+
+            OpenApiSchema colSchema;
+            if (context.Settings.EnablePagination && context.Settings.EnableCount)
+            {
+                colSchema = new OpenApiSchema
+                {
+                    AllOf = new List<OpenApiSchema>
+                    {
+                        paginationSchema,
```

Comment:
this one with 3 allOf is going to be challenging since it suggests multi-inheritance, which most languages don't support.
My suggestion it to make it easier for code generators by making count an all of inline count definition + reference to pagination (or vice versa) when both settings are enabled.
And here we'd only be left with the last two blocks and the following condition for the first block
`context.Settings.EnablePagination && !context.Settings.EnableCount`
What do you think?

Reply:
Or alternatively, we can have a 3rd `BaseCollectionPaginationCountResponse` schema that combines the two. When both settings are `true` then we use the above reference schema, else we use the individual respective schemas.

Reply:
and that new base would redefine at least one property, not be a schema with 2 referenced all of entries right?

Reply:
The entity collection would look like this:

```
microsoft.graph.userCollectionResponse:
  title: Collection of user
  type: object
  allOf:
    - $ref: '#/components/schemas/BaseCollectionPaginationCountResponse'
    - type: object
      properties:
        value:
          type: array
          items:
            $ref: '#/components/schemas/microsoft.graph.user'
```

and the new combined base `BaseCollectionPaginationCountResponse` would look like:

```
BaseCollectionPaginationCountResponse:
  title: Base collection pagination and count responses
  type: object
  properties:
    '@odata.nextLink':
      type: string
    '@odata.count':
      type: integer
      format: int64
```
So in total, we have 3 collection bases:
1. Pagination
2. Count
3. Pagination + Count

Each will be defined depending on the settings

Reply:
alright, let's roll with that. Thanks for the discussion.